src/connectors/voice-search/__tests__/connectVoiceSearch-test.js

```diff
@@ -65,6 +65,28 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/voice-searc
         })
       );
     });
+
+    it('creates custom voice helper', () => {
+      const voiceHelper = {
+        isBrowserSupported: () => true,
+        dispose: () => {},
+        getState: () => ({
+          isSpeechFinal: true,
+          status: 'askingPermission',
+          transcript: '',
+        }),
+        isListening: () => true,
+        toggleListening: () => {},
```

Comment:
If we make this API public, the options should be explicit. I'd suggest remplacing `toggleListening` by `onListenStart` and `onListenEnd`.

Reply:
toggle listening is a function to toggle the listening, not event listeners

Reply:
Than perhaps `start` and `stop`.

It seems like `toggleListening` is redundant with `dispose` too, since it calls `dispose` despite `dispose` being exported too.

Reply:
dispose removes listeners, so it's different. We could expose start & stop from the helper & put toggleListening inside the connector indeed